docs/events.md

```diff
@@ -0,0 +1,165 @@
+# Events
+
+> v2.11 and after
+
+![alpha](assets/alpha.svg)
+
+## Overview
+
+To support external webhooks, we have this endpoint `/api/v1/events/{namespace}`. Events can be sent to that can be any JSON data.
+
+These events can submit a workflow template (cluster workflow templates are not supported today).
+
+You may also wish to read about [webhooks](webhooks.md).
+
+## Authentication and Security
+
+Clients wanting to send events to the endpoint need an [access token](access-token.md).   
+
+It is only possible to submit workflow templates your access token has access to, [example](manifests/quick-start/base/webhooks/submit-workflow-template-role.yaml).
+
+Example (note the trailing slash):
+
+```bash
+curl https://localhost:2746/api/v1/events/argo/ \
+  -H "Authorization: $ARGO_TOKEN" \
+  -d '{"message": "hello"}'
+```
+
+With a **discriminator**:
+
+```bash
+curl https://localhost:2746/api/v1/events/argo/my-discriminator \
+  -H "Authorization: $ARGO_TOKEN" \
+  -d '{"message": "hello"}'
+```
+
+The event endpoint will always return in under 10 seconds because the event will be queued and processed asynchronously. This means you will not be notified synchronously of failure. It will only return a failure (503) if the event processing queue is full.  
```

Comment:
Do we not also validate and reject unreadable payloads?

Reply:
payloads can be any JSON - non-JSON would be rejected

Reply:
Change to `It will return a failure (503) if the event processing queue is full. `